SU2_CFD/src/drivers/CDriver.cpp

```diff
@@ -2134,6 +2171,58 @@ void CDriver::Numerics_Preprocessing(CConfig *config, CGeometry **geometry, CSol
       numerics[iMGlevel][TRANS_SOL][conv_bound_term] = new CUpwLin_TransLM(nDim, nVar_Trans, config);
     }
   }
+  
+  /*--- Solver definition for the scalar transport model problem ---*/
+  
+  if (is_scalar_model) {
+    
+    /*--- Definition of the convective scheme for each equation and mesh level ---*/
+    
+    switch (config->GetKind_ConvNumScheme_Scalar()) {
+      case NONE :
+        break;
+      case SPACE_UPWIND :
+        for (iMGlevel = 0; iMGlevel <= config->GetnMGLevels(); iMGlevel++) {
+          if (passive_scalar || progress_variable) {
+            numerics[iMGlevel][SCALAR_SOL][CONV_TERM] = new CUpwSca_transportedScalar_general(nDim, nVar_Scalar, config);
+          }
+        }
+        break;
+      default :
+        SU2_MPI::Error("Convective scheme not implemented (scalar transport).", CURRENT_FUNCTION);
+        break;
+    }
+    
+    /*--- Definition of the viscous scheme for each equation and mesh level ---*/
+    
+    for (iMGlevel = 0; iMGlevel <= config->GetnMGLevels(); iMGlevel++) {
+      if (passive_scalar || progress_variable){
+        if (iMGlevel == MESH_0) numerics[iMGlevel][SCALAR_SOL][VISC_TERM] = new CAvgGrad_transportedScalar_general(nDim, nVar_Scalar, true, config);
+        else numerics[iMGlevel][SCALAR_SOL][VISC_TERM] = new CAvgGrad_transportedScalar_general(nDim, nVar_Scalar, false, config);
+      }
+    }
+    
+    /*--- Definition of the source term integration scheme for each equation and mesh level ---*/
+    
+    for (iMGlevel = 0; iMGlevel <= config->GetnMGLevels(); iMGlevel++) {
+      if (passive_scalar || progress_variable) {
+        numerics[iMGlevel][SCALAR_SOL][source_first_term]  = new CSourcePieceWise_transportedScalar_general(nDim, nVar_Scalar, config);
+   //if (config->GetAxisymmetric() == YES)
+      //    numerics[iMGlevel][SCALAR_SOL][SOURCE_SECOND_TERM] = new CSourceAxisymmetric_Scalar(nDim, nVar_Scalar, config);
+      //else  
+        numerics[iMGlevel][SCALAR_SOL][SOURCE_SECOND_TERM] = new CSourceNothing(nDim, nVar_Scalar, config);
```

Comment:
Use the lower case variable `source_second_term` otherwise this will leak memory when using OpenMP and then segfault.
We instantiate one numerics per thread.